### PR TITLE
fix: catch undefined LogicalName when Lambda Layer versions are deleted from the console

### PR DESCRIPTION
**Pull request by sdstolworthy**
#### Description of changes
Provides a more descriptive error when attempting to use a Lambda Layer that was deleted in the AWS Console.

#### Description of how you validated changes
New unit test added.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/lambda-layer-cloudformation-template.ts [hunk 1]
```diff
@@ -30,7 +30,12 @@ export function generateLayerCfnObj(isNewVersion: boolean, parameters: LayerPara
     layerCloudState.latestVersionLogicalId = logicalName;
     versionList.unshift({ LogicalName: logicalName, legacyLayer: false });
   } else {
-    logicalName = _.first(versionList).LogicalName;
+    logicalName = _.first(versionList)?.LogicalName;
+    if (!logicalName) {
+      throw new AmplifyError('LambdaLayerNotFoundError', {
+        message: 'No versions were found for the Lambda Layer. Were they deleted?',
```
Comment (sobolk) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/lambda-layer-cloudformation-template.ts:36]: if this refers to layers stored in the cloud we should somehow convey where the deletion could have happen.
I'm assuming the deletion must have happened outside of amplify - should that be part of the question?

**Review by sobolk — APPROVED** [review]

**Comment by sdstolworthy** [comment]
Reverting to draft. I need to do more investigation on this issue.

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/lambda-layer-cloudformation-template.ts (hunk 1 above)
Comment (sdstolworthy) [reply packages/amplify-category-function/src/provider-utils/awscloudformation/utils/lambda-layer-cloudformation-template.ts]: Tried to make the message a bit clearer. I was not able to reproduce the specific error, so I can't give much better guidance than to check the console at this point. I'm open to suggestions on messaging.

**Review by jhockett — DISMISSED** [review]

File: packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerCloudState.ts [hunk 2]
```diff
@@ -103,6 +103,11 @@ export class LayerCloudState {
         // eslint-disable-next-line no-param-reassign
         layerVersion.legacyLayer = layerVersion.LogicalName === undefined || layerVersion.LogicalName === 'LambdaLayer';
       });
+      if (layerVersionList.length === 0) {
+        throw new AmplifyError('LambdaLayerNotFoundError', {
+          message: 'No versions were found for the Lambda Layer. Were they deleted on the AWS Lambda Console?',
+        });
+      }
```
Comment (Amplifiyer) [inline packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerCloudState.ts:110]: Should we do this before line 45 in this file? Maybe also add the layer name and environment in the error message?

Comment (sdstolworthy) [reply packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerCloudState.ts]: Good suggestions. The error message now includes the name of the missing layers.

**Review by Amplifiyer — APPROVED** [review]

**Review by jhockett — APPROVED** [review]

**Pull request merged**